### PR TITLE
Feature/GitHub list commands

### LINKED ISSUE #25: [User Story] As a user, I want a command that stores user's GitHub usernames to that users are easily identifiable
**Given...**
A user exists on the server that is in the GitHub Organization

**When...**
The current user sends the command "c!finduser @DiscordUser"

**Then...**
Reply with the Discord User's GitHub username

### LINKED ISSUE #57: [User Story] As a new recruit, I want to know what all the teams that are in the organization.
**Given...**
I'm in the #bot channel.

**When...**
I send a command like `c!github list-team`

**Then...**
Bot replies by showing all the teams that are in the organization.

### PR DESCRIPTION
**Pull request by nickmartin1ee7**
New commands
* github list-org - Lists organization members
* github list-teams - Lists all organization's teams
* github list-team-members [team-slug] - Lists all team members

The new commands have shorthand aliases as well.

Disables deprecated commands:
* join
* create

### CODE REVIEW

**Comment by nickmartin1ee7** [comment]
Sample output
image
image

File: Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinOrgLogic.cs [hunk 1]
```diff
@@ -41,11 +41,15 @@ public override async Task<Embed> ExecuteAsync()
...
 
-        private async Task<EmbedContent> JoinGitHubOrganization()
+        private async Task<HttpResponseMessage> PostNewInviteToOrganization()
         {
-            var githubClient = new GitHubClient(_githubAuthToken);
-            var response = await githubClient.InviteViaEmailToOrganizationAsync(_githubOrganizationName, _userGitHubEmail);
-            return ParseHttpResponseToEmbedContent(response);
+            var githubClient = new GitHubClient(_githubAuthToken, _githubOrganizationName);
+
+            var json = $"{{\"email\":\"{_userGitHubEmail}\"}}";
```
Comment (srjheam) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinOrgLogic.cs]: I think we maybe should avoid this "magic string" and start to use `Json.SerializeObject` to return this json string.
```csharp
var data = new Dictionary<object, object>
{
   ["email"] = "sample@example.com"
};

var json = JsonConvert.SerializeObject(data);
```

File: Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinTeamLogic.cs [hunk 2]
```diff
@@ -42,11 +38,13 @@ public override async Task<Embed> ExecuteAsync()
             return embed;
         }
 
-        private async Task<EmbedContent> JoinGitHubTeam()
+        private async Task<HttpResponseMessage> PutNewMemberToTeamAsync()
         {
-            var githubClient = new GitHubClient(_githubAuthToken);
-            var response = await githubClient.AddOrUpdateTeamMembershipForUserAsync(_githubOrganizationName, _teamSlug, _username, Team.Role.member);
-            return ParseHttpResponseToEmbedContent(response);
+            const string json = "{\"role\":\"member\"}";
```
Comment (srjheam) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinTeamLogic.cs]: Start using `Json.SerializeObject` to return this string?

File: Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinOrgLogic.cs (hunk 1 above)
Comment (nickmartin1ee7) [reply Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinOrgLogic.cs]: Good suggestion! I'll look into updating it. Feel free to mark your review as "request changes"

File: Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubListOrgLogic.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                var count = orgMembersJToken.Count();
+
+                descriptionSb.Append($"**Total Members in Organization: {count}**{Environment.NewLine}");
+
+                for (var i = 0; i < count; i++)
+                {
+                    descriptionSb.Append($"{i + 1}: {orgMembersJToken[i]["login"]} ({orgMembersJToken[i]["url"]}){Environment.NewLine}");
+                }
+
+                embedContent.Title += "Successful";
```
Comment (srjheam) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubListOrgLogic.cs]: Do we really need to tell that to the user? I think it'd be better if we just send a message communicating the user in the event of a failure.

File: Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubListOrgLogic.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                embedContent.Description = descriptionSb.ToString();
+            }
+            else
+            {
+                var errMsg =
+                    $"There was an error requesting to list the Organization's Members! GitHub's reason was: {response.ReasonPhrase}";
+
+                _logger.Error(errMsg);
+
+                embedContent.Title += "Failed";
```
Comment (srjheam) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubListOrgLogic.cs:75]: Maybe just that would be better, what do you think?
````csharp
embedContent.Title = "Oops! An error has occurred";

File: GitHubApiWrapper/GitHubClient.cs [hunk 5]
```diff
@@ -1,58 +1,20 @@
...
-using System.Threading.Tasks;
-using GitHubApiWrapper.Models;
-using GitHubApiWrapper.Models.CustomExceptions;
 
 namespace GitHubApiWrapper
 {
     public class GitHubClient
     {
-        private readonly HttpClient _client;
+        public readonly HttpClient Client;
```
Comment (srjheam) [inline GitHubApiWrapper/GitHubClient.cs]: Exposing the `HttpClient` allows the developer to overwrite the authentication token, possibly causing annoying issues in the future.

Comment (nickmartin1ee7) [reply GitHubApiWrapper/GitHubClient.cs:9]: It's readonly, and has no set accessor

Comment (nickmartin1ee7) [reply GitHubApiWrapper/GitHubClient.cs]: It's readonly, so they shouldn't be able to set a new instance to it

File: Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubListOrgLogic.cs (hunk 3 above)
Comment (nickmartin1ee7) [reply Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubListOrgLogic.cs]: I'll look into it. I'll have to change that pattern for the other github commands.

File: Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinTeamLogic.cs (hunk 2 above)
Comment (nickmartin1ee7) [reply Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinTeamLogic.cs]: Sounds good

File: GitHubApiWrapper/GitHubClient.cs (hunk 5 above)
Comment (srjheam) [reply GitHubApiWrapper/GitHubClient.cs]: But the property `Client.DefaultRequestHeaders.Authorization` isn't and that may be a problem.

Comment (nickmartin1ee7) [reply GitHubApiWrapper/GitHubClient.cs]: I see. Good point. I can hide the underlying client and expose public methods instead.

File: Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinOrgLogic.cs (hunk 1 above)
Comment (srjheam) [reply Code2Gether-Discord-Bot.Library/BusinessLogic/GitHubJoinOrgLogic.cs]: > Feel free to mark your review as "request changes"

One of my first times reviewing a PR, so I did it all wrong LFMAO

File: GitHubApiWrapper/GitHubClient.cs (hunk 5 above)
Comment (srjheam) [reply GitHubApiWrapper/GitHubClient.cs]: In my view, that would be good approach.

**Comment by nickmartin1ee7** [comment]
@srjheam Made changes with suggestions.

**Review by srjheam — APPROVED** [review]
LGTM

**Pull request merged**